app/views/govuk_publishing_components/components/_task_list.html.erb

```diff
@@ -2,102 +2,97 @@
   options = {}
   options[:heading_level] = heading_level ||= 2
 
-  groups ||= false
+  steps ||= false
   small ||= false
   show_step ||= false
   remember_last_step ||= false
   task_list_url ||= false
   task_list_url_link_text ||= "Get help completing this step"
-  highlight_group ||= false
+  highlight_step ||= false
 
   step_count = 0
+  step_number = 0
 %>
-<% if groups %>
+<% if steps %>
   <div
     data-module="gemtasklist"
     class="gem-c-task-list js-hidden <% unless small %>gem-c-task-list--large<% end %>"
     <% if remember_last_step %>data-remember<% end %>
-    >
-    <ol class="gem-c-task-list__groups">
-      <% groups.each_with_index do |group, group_index| %>
+  >
+    <ol class="gem-c-task-list__steps">
+      <% steps.each_with_index do |step, step_index| %>
         <%
-          group_is_active = group_index + 1 == highlight_group
+          step_is_active = step_index + 1 == highlight_step
+          step_count += 1
+          id = TaskListHelper.new.generate_task_list_id(step[:title])
+
+          logic = false
```

Comment:
This may need renaming - it's not obvious what `logic` is referring to, and it's difficult to get any kind of context when it's used, e.g:

```
<% if logic %>
    <%= logic %>
```

Reply:
I'll make a note to look at this in a separate PR, but this is outside the scope of this change. Good point though.

Reply:
I'm OK with `logic` for the moment, as we've been trying to think of something else that describes it better and haven't quite managed it yet.  

Our schema (https://github.com/alphagov/govuk-content-schemas/blob/25c0bc646225dfde0389861d1f62bb82bfb498ac/formats/shared/definitions/_step_by_step_nav.jsonnet#L29-L35) has it limited to `and` or `or` and it's used to interrupt the usual numbering sequence to indicate a concurrent step.

Reply:
@sihugh Sure, I don't think it needs to block this PR, but I think it's worth changing in future as it's not very easily understandable if other people come to work on this.